UndertaleModLib/Util/GMImage.cs

```diff
@@ -335,6 +345,14 @@ public static GMImage FromBinaryReader(IBinaryReader reader, long maxEndOfStream
             return FromQoi(reader.ReadBytes(12 + (int)compressedLength));
         }
 
+        // DDS
+        if (header.StartsWith(MagicDds))
+        {
+            // Read entire image
+            reader.Position = startAddress;
+            return FromDds(reader.ReadBytes((int)(maxEndOfStreamPosition - startAddress)));
```

Comment:
If possible, we should be finding the length of the image data using the image contents, rather than using `maxEndOfStreamPosition`, which can result in unnecessary padding when re-saving with modifications.

Reply:
Didn't think about that! Sadly there's no easy way to find the length in this format. I can get the size of file without the extra bdata2 field, because getting that is even harder.